File: tests/javascript/simple.js
import { readFile } from 'fs';

class Color {
  constructor(name) {
    this.name = name;
  }
}

function factorial(n) {
  if (n === 0) {
    return 1;
  }
  return n * factorial(n - 1);
}

const msg = "hello";
const x = 42;
const b = true;
console.log(factorial(x));

File: tests/haskell/simple.hs
module Main where

import Data.List

factorial :: Int -> Int
factorial 0 = 1
factorial n = n * factorial (n - 1)

data Color = Red | Green | Blue

type Name = String

main :: IO ()
main = do
  let x = 42
  let msg = "hello"
  print (factorial x)
